### PR TITLE
Dynamic format

### PR DESCRIPTION
**Pull request by Demonkratiy**
Added capabilities and logic to support dynamic format.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [14.x]
+        node-version: [15.x]
```
[the remark this answers is not in the document]
Comment (Demonkratiy) [reply .github/workflows/build.yml]: done

File: package.json [hunk 2]
```diff
@@ -1,26 +1,29 @@
 {
     "name": "visual",
-    "version": "3.2.0",
+    "version": "4.0.0",
```
[the remark this answers is not in the document]
Comment (Demonkratiy) [reply package.json:3]: done

File: src/barChart.ts [hunk 3]
```diff
@@ -428,16 +429,20 @@ export class BarChart implements IVisual {
             <ISelectionId[]>this.selectionManager.getSelectionIds()
         );
 
-        barSelectionMerged.on('click', (d) => {
+        barSelectionMerged.on('click', (event) => {
```
[the remark this answers is not in the document]
Comment (Demonkratiy) [reply src/barChart.ts]: done

File: src/barChart.ts [hunk 4]
```diff
@@ -428,16 +429,20 @@ export class BarChart implements IVisual {
...
             // Allow selection only if the visual is rendered in a view that supports interactivity (e.g. Report)
             if (this.host.hostCapabilities.allowInteractions) {
-                const isCtrlPressed: boolean = (<MouseEvent>d3Event).ctrlKey;
+                const isCtrlPressed: boolean = (<MouseEvent>event).ctrlKey;
+                const mouseEvent: MouseEvent = event;
+                const eventTarget: EventTarget = mouseEvent.target;
+
+                const dataPoint: any = d3Select(<d3.BaseType>eventTarget).datum();
```
[the remark this answers is not in the document]
Comment (Demonkratiy) [reply src/barChart.ts]: done

**Comment by MulyukovAidar** [comment]
LGTM

**Pull request merged**